### PR TITLE
DRY out default stage info handling

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

**Review by JohnMalmberg — APPROVED** [review]

**Comment by brianjmurrell** [comment]
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-7420/5/pipeline/ is a "current master" test run demonstrating this PR not causing any regressions.

File: vars/scanRpms.groovy [hunk 1]
```diff
@@ -65,7 +65,7 @@ def call(Map config = [:]) {
 
   provisionNodes NODELIST: nodelist,
                  node_count: 1,
-                 distro: stage_info['ci_target'],
+                 distro: (stage_info['ci_target'] =~ /([a-z]+)(.*)/)[0][1] + stage_info['distro_version'],
```
Comment (JohnMalmberg) [inline vars/scanRpms.groovy:68]: This disables using using a pragma or command line parameter to specify an test CI image.

File: vars/storagePrepTest.groovy [hunk 2]
```diff
@@ -64,7 +64,7 @@ def call(Map config = [:]) {
 
   provisionNodes NODELIST: nodelist,
                  node_count: stage_info['node_count'],
-                 distro: stage_info['ci_target'],
+                 distro: (stage_info['ci_target'] =~ /([a-z]+)(.*)/)[0][1] + stage_info['distro_version'],
```
Comment (JohnMalmberg) [inline vars/storagePrepTest.groovy:67]: Same as above.

File: vars/unitTest.groovy [hunk 3]
```diff
@@ -84,7 +84,7 @@ def call(Map config = [:]) {
 
   provisionNodes NODELIST: nodelist,
                  node_count: stage_info['node_count'],
-                 distro: stage_info['ci_target'],
+                 distro: (stage_info['ci_target'] =~ /([a-z]+)(.*)/)[0][1] + stage_info['distro_version'],
```
Comment (JohnMalmberg) [inline vars/unitTest.groovy:87]: Same as above.

**Review by JohnMalmberg — APPROVED** [review]

File: vars/scanRpms.groovy (hunk 1 above)
Comment (brianjmurrell) [reply vars/scanRpms.groovy:68]: Probably.  All of the target handling needs to be refactored to deal with current requirements.  Mainly this is just more fallout from having to be able to support multiple (EOL, unsupported even) point releases.

File: vars/storagePrepTest.groovy (hunk 2 above)
Comment (brianjmurrell) [reply vars/storagePrepTest.groovy:67]: Ditto.

File: vars/unitTest.groovy (hunk 3 above)
Comment (brianjmurrell) [reply vars/unitTest.groovy:87]: Ditto.

**Pull request merged**